source/guides/basics/loops.md

```diff
@@ -46,7 +139,74 @@ my_move_task:
+
+As a word of warning, `while` loops should be avoided if possible. It is very easy to make what is known as an infinite loop, as in the loop will never have a chance to stop. Once its running, it is very hard to force it to stop beyond just stopping the server.
+
+### Stop The Loop
+
+Sometimes in loops you only want to keep looping until you reach a certain point, however not necessarily stop the script. While `stop` is used to stop scripts entirely, `repeat/foreach/while stop` is used to stop the relevant loop
+
+For instance in the first example we had, if we wanted to stop sending lightning if the player had less than 5 health we would do as follows
+
+```dscript_green
+my_lightning_task:
+    type: task
+    script:
+    - repeat 5 as:count:
+        - strike <player.location>
+        - narrate "you have been struck by lightning <[count]> times!"
+        - if <player.health> < 5:
+            - repeat stop
+        - wait 1s
+    - narrate "no more lightning"
+```
+Take note, only the loop stops, the end `narrate` line still runs.
+
+It can also be used in `foreach` loops. For instance, if we wanted to stop looping if we found a baby cow in the example above
+
+```dscript_green
+my_cow_task:
+    type: task
+    script:
+    - foreach <player.location.find_entities[cow].within[30]> as:cow:
+        - narrate "<[loop_index]> - There's a cow just <[cow].location.distance[<player.location>].round> blocks away!"
+        - playeffect effect:fireworks_spark at:<[cow].location> visibility:50 quantity:100 data:0 offset:3
+        - if <[cow].is_baby>:
+            - narrate "omg its a baby!!"
+            - foreach stop
+```
+
+### Next Please
+
+There are times where you want to skip to the next loop without running the rest of the current loop. For that you use `repeat/foreach/while next`
+
+Take the `foreach` example we had, lets say we want to skip over all baby cows. While you can put the whole block inside an `if` command like so...
+```dscript_green
+my_cow_task:
+    type: task
+    script:
+    - foreach <player.location.find_entities[cow].within[30]> as:cow:
+        - if !<[cow].is_baby>:
+            - narrate "<[loop_index]> - There's a cow just <[cow].location.distance[<player.location>].round> blocks away!"
+            - playeffect effect:fireworks_spark at:<[cow].location> visibility:50 quantity:100 data:0 offset:3
+```
+
+You could instead use `next`
+
+```dscript_green
+my_cow_task:
+    type: task
+    script:
+    - foreach <player.location.find_entities[cow].within[30]> as:cow:
+        - if <[cow].is_baby>:
+            - foreach next
+        - narrate "<[loop_index]> - There's a cow just <[cow].location.distance[<player.location>].round> blocks away!"
+        - playeffect effect:fireworks_spark at:<[cow].location> visibility:50 quantity:100 data:0 offset:3
+```
+
```

Comment:
This isn't a great example, as an if statement probably works better here.

Reply:
does it need to be a great example? Just needs to get the point across...

Reply:
This segment is a demonstration of the usefulness of the next argument, so yes the examples should make sense in my opinion